File: src/p0070/cpp/solution.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n == 1) {
            return 1;
        } else if (n == 2) {
            return 2;
        }

        int ways = 0, i = 2, x = 1, y = 2;
        while (i++ < n) {
            ways = x + y;
            x = y;
            y = ways;
        }

        return ways;
    }
};

File: src/p0070/python/solution.py
class Solution:
    def climbStairs(self, n: int) -> int:
        pre, cur = 1, 2
        if n == 2:
            return cur
        elif n == 1:
            return pre
        i = 2
        while i < n:
            temp = cur
            cur += pre
            pre = temp
            i += 1
            
        return cur